src/compiler/checker.ts

```diff
@@ -28838,8 +28838,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     // it is the type of the numeric index signature in T. Otherwise, in ES6 and higher, the contextual type is the iterated
     // type of T.
     function getContextualTypeForElementExpression(arrayContextualType: Type | undefined, index: number): Type | undefined {
+        const hasIterable = getGlobalIterableType(/* reportErrors */ false) !== emptyGenericType;

```

Comment:
This part ain't final - it probably should look slightly different. I couldn't find, at the moment, an easier way to check if `Iterable` is available + I didn't find any function that would check if something is assignable to `Iterable<any>`. 

Do you know if there is any code like this that I missed? Or should I add it?

Reply:
Can it be done in the same way we check if some thing's an array? I believe that also checks for assignability to Array<any>

Reply:
I'm not entirely sure if we think about the same thing. I can just move this iterable-related stuff to a function like `isIterable`. Then this filtering would look smth like this:
```diff
-filterType(arrayContextualType, t => isArrayOrTupleLikeType(t) || !!getIndexTypeOfType(t, numberType) || hasIterable && isTypeAssignableTo(t, createIterableType(anyType)))
+filterType(arrayContextualType, t => isArrayOrTupleLikeType(t) || !!getIndexTypeOfType(t, numberType) || isIterable(t))
```

Reply:
Yes, sorry, it's early and I didn't scroll over enough to see the code 😅

Reply:
I pushed out some cleanup - introduced `isAssignableToAvailableAnyIterable` and reused that in another place that is checking the same thing.